app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -1239,4 +1247,21 @@ void openImageInGallery(@NonNull String language) {
         leadImagesHandler.openImageInGallery(language);
     }
 
+    private void maybeShowFundraisingDialog() {
+        if (Prefs.hasVisitedArticlePage() && Prefs.fundraisingDialogShownInYear() < Calendar.getInstance().get(Calendar.YEAR)) {
+            disposables.add(ServiceFactory.getRest(getTitle().getWikiSite()).getAnnouncements()
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(list -> {
+                        String country = GeoUtil.getGeoIPCountry();
+                        Date now = new Date();
+                        for (Announcement announcement : list.items()) {
+                            if (shouldShow(announcement, country, now) && announcement.placement().equals(FUNDRAISING_PLACEMENT_ARTICLE)) {
+                                new AnnouncementDialog(requireActivity(), new AnnouncementCard(announcement)).show();
```

Comment:
We should probably add `setCancelable(false)` so that the user must choose one of the options in the dialog, and won't dismiss it with an accidental tap outside the dialog area.

Reply:
Sure!